proposals/public-suffix.md

```diff
@@ -0,0 +1,555 @@
+| #2   | bar.amazonaws.com           | amazonaws.com               | ICANN       | No           |
+|      |                             |                             |             |              |
+| #1   | foo.amazonaws.com           | amazonaws.com               | ICANN       |              |
+| #2   | bar.us-east-1.amazonaws.com | bar.us-east-1.amazonaws.com | Private     | Yes          |
+
+##### PSL Requirements
+
+| PSL Feature            | Requirement | Discussion |
+|------------------------|-------------|------------|
+| Allow Private Suffixes | Yes         | including all suffixes in PSL means more information about third-party boundaries |
+| Allow Unknown Suffixes | Yes         | provides better performance |
+| Preserve IDN Unicode   | No          | provides better performance |
+
+### Known Consumers
+
+Mozilla intends to make use of this API in its [multi-account-containers](https://addons.mozilla.org/en-US/firefox/addon/multi-account-containers/)
+extension to allow users to create containers that isolate organizational base domains:
+[PR #2352](https://github.com/mozilla/multi-account-containers/pull/2352).
+Currently, users must manually curate their containers to capture all anticipated subdomains
+of an organization's base domain, and this manual approach is cumbersome and error-prone.
+
+In addition to this, an analysis has been done of the relevant use cases of a selection of
+other extensions that currently roll their own solutions in order to parse the PSL, and the
+use cases are set out in the [Use Cases](#use-cases) section. Although the extension
+authors have not been consulted directly, it is assumed that they would stand to benefit
+from this API due to the reasons stated in the [Objective](#objective) section.
+
+| Extension | Relevant use cases | Description |
+|-----------|--------------------|-------------|
+| [multi-account-containers](https://addons.mozilla.org/en-US/firefox/addon/multi-account-containers/) | #1, #2 | isolate requests into containers by organizational base domain, view all domains grouped by organization |
+| [uBlock Origin](https://addons.mozilla.org/en-US/firefox/addon/ublock-origin/) | #1, #2, #3 | create organization/3rd-party request-blocking rules, view request blocking status by organization, create blocking rules for specifically 3rd-party requests |
+| [adblock-plus](https://addons.mozilla.org/en-US/firefox/addon/adblock-plus/) | #1, #3 | create organization/3rd-party request-blocking rules |
+| [duckduckgo-for-firefox](https://addons.mozilla.org/en-US/firefox/addon/duckduckgo-for-firefox/) | #3 | exclude requests from tracking detection unless they are 3rd-party |
+| [violentmonkey](https://addons.mozilla.org/en-US/firefox/addon/violentmonkey/) | #1 | run scripts automatically for requests having a specific organization name plus any suffix |
+
+## Specification
+
+### Schema
+
+A new API `publicSuffix` is added as follows:
```

Comment:
I've dug into PSL use-cases a couple of time now. I think it might be better to make the API
`isValidURL(...)` and `isSameSite(...)`. That should be enough for use-cases (1.) and (3.) above. For use case (2.) you're restricted to comparison based sorting algorithms but that's probably okay in practice.

Reply:
As @Rob--W just pointed out in chat, there are some user-interface use-cases which would still need the `getRegistrableDomain` function. For example, highlighting in the URL bar or giving a name to a group of same-site URLs.

I still think it would be good to have an explicit API for the two cases mentioned above.